crates/ruff_db/src/diagnostic/render.rs

```diff
@@ -820,6 +828,248 @@ fn relativize_path<'p>(cwd: &SystemPath, path: &'p str) -> &'p str {
     path
 }
 
+/// A measure of the width of a line of text.
+#[derive(Clone, Copy, Default)]
+struct LineWidthBuilder {
```

Comment:
The name here feels a bit off now, considering that it isn't a proper builder anymore.

Maybe just call it `Position` and I suggest renaming `get` to `width` to make it clear what it returns (or remove the accessor all together and access the field directly)